doc/source/sphinxext/postprocess_api.py

```diff
@@ -0,0 +1,56 @@
+"""
+Sphinx plugin to run example scripts and create a gallery page.
+
+Lightly modified from the seaborn project (Michael Waskom).
+Originally, lightly modified from the mpld3 project.
+
+"""
+import os.path as op
+import glob
+from sphinx.application import Sphinx
+
+
+def on_build_finished(app: Sphinx, exception: Exception) -> None:
+    html_dir = op.abspath(app.builder.outdir)
+    api_dir = op.join(html_dir, 'api')
+
+    # Check if the index has Jekyll template marks
+    has_jekyll_marks = False
+    index_html = op.join(html_dir, 'index.html')
+    with open(index_html, 'rt') as f:
+        lines = f.readlines()
+        if sum(line == '---\n' for line in lines) == 2:
+            has_jekyll_marks = True
+        mark_start = lines.index('---\n')
```

Comment:
Won't this throw an error if the output has no such marks? Or is that exactly what we want?

Reply:
The logic will change a bit, but if the `index.html` starts with a `---\n` it's a Jekyll template, and indeed should have another one of those lines (signifying  `</head>` to Jekyll). If you don't have the second line, i'ts like having an HTML file without closing its head... we should not let this error through.